spec/integration/util/purageable_spec.rb

```diff
@@ -0,0 +1,41 @@
+require 'spec_helper'
+require 'r10k/util/purgeable'
+require 'r10k/util/cleaner'
+
+require 'tmpdir'
+
+RSpec.describe R10K::Util::Purgeable do
+  it 'purges only unmanaged files' do
+    Dir.mktmpdir do |envdir|
+      managed_directory = "#{envdir}/managed_one"
+      desired_contents = [
+        "#{managed_directory}/expected_1",
+        "#{managed_directory}/managed_subdir_1",
+        "#{managed_directory}/managed_symlink_dir",
+        "#{managed_directory}/managed_subdir_1/subdir_expected_1",
+        "#{managed_directory}/managed_subdir_1/managed_symlink_file",
+      ]
+
+      FileUtils.cp_r('spec/fixtures/unit/util/purgeable/managed_one/',
+                     managed_directory)
+
+      cleaner = R10K::Util::Cleaner.new([managed_directory], desired_contents)
+
+      cleaner.purge!({ recurse: true, whitelist: ["**/subdir_allowlisted_2"] })
+
+      # Files present after purge
+      expect(File.exist?("#{managed_directory}/expected_1")).to be true
+      expect(File.exist?("#{managed_directory}/managed_subdir_1")).to be true
+      expect(File.exist?("#{managed_directory}/managed_symlink_dir")).to be true
+      expect(File.exist?("#{managed_directory}/managed_subdir_1/subdir_expected_1")).to be true
+      expect(File.exist?("#{managed_directory}/managed_subdir_1/managed_symlink_file")).to be true
```

Comment:
might be nice to just do
```suggestion
      desired_contents.each do |path|
        expect(File.exist?(path)).to be true
      end
```

Reply:
I had that first and the problem is that if it fails, all you get is:
```
Failure/Error: expect(File.exist?(path)).to be true

expected true
got false
```
Which is utterly useless if there's a failure.

Reply:
ugh that's so annoying